objectRecognitionSource/src/main/java/edu/ml/tensorflow/ObjectDetector.java

```diff
@@ -74,12 +81,35 @@
      * @param vantiq        The Vantiq variable used to connect to the VANTIQ SDK. Either authenticated, or set to null.
      * @param sourceName    The name of the VANTIQ Source
      */
-    public ObjectDetector(float thresh, String graphFile, String labelFile, double[] anchorArray, ImageUtil imageUtil, String outputDir, 
+    public ObjectDetector(float thresh, String graphFile, String labelFile, String metaFile, double[] anchorArray, ImageUtil imageUtil, String outputDir, 
             Boolean labelImage, int saveRate, Vantiq vantiq, String sourceName) {
         try {
             GRAPH_DEF = IOUtil.readAllBytesOrExit(graphFile);
-            LABELS = IOUtil.readAllLinesOrExit(labelFile);
-            this.anchorArray = anchorArray;
+            // Parse meta file if it exists, and either the label file or anchor config options have not been set
+            if (metaFile != null && (labelFile == null || anchorArray == null)) {
+                parseMetaFile(metaFile);
```

Comment:
Seems like we should parse the metaFile regardless of presence of labelFile or anchorArray.  No reason not too & we may use something from it in the future.

Reply:
Sure I can add that. Only reason I wasn't doing it was to save memory.

I'll change the condition to just check that the metaFile != null.